### ISSUE TITLE
VAE model training loss function

### ISSUE THREAD

**Issue by Adnan-annan** [masato-ka/airc-rl-agent#28]
When I am training van model for a different track than you, my loss is very big, like 3196517 even after 800 epochs. Is it normal? and the reconstruction results is not satisfactory also. I collected around 70000 images for training. can you guide me through training process of vae ?

**Comment by Adnan-annan** [comment]
hi @masato-ka  i am facing trouble training VAE model. Kindly guide me towards this training process. i will be very thanful to you.

**Comment by masato-ka** [comment]
Usually loss value is very high. For example, loss value started 1658790.625 at  training that using 10k datasets. I think initial loss value is increase with dataset size. When loss value decreasing with progressing epoch, training is success. You can confirm loss value and reconstruction image on TensorBoards. Reconstruction image is very lazy, because this vae model is beta-VAE. If you needs clear image, You can change beta-value in loss_fn method.
modify 50.0 beta value to 0.0.
'''   
# def loss_fn last line.
        return reconstruction + 50.0 * KL
'''

**Comment by Adnan-annan** [comment]
Actually i had few questions about this training process:
1)  During the collection of data, i move the jetracer along the track by hand. First one complete circle in the center of track. and one circle while moving car in left side along the track , and one circle while moving car in right side along the track. In your opinion is it right way to collect the dataset? I need to have confidence on this step to really focus on training VAE. Would you like to share your training data for reference ?

2)  After training on alot of epochs, the model didnt really learn to differentiate between black (no image) and image of the track. which seems to me that learning did not happen.  what suggestions would you give me for better result?

3)  I tried training the model on 1000 images of image of just a same place in the middle of track. when i check the results, no matter what is the input image, model just reproduces the same image. I dont quite understand this phemnomenon.

**Comment by masato-ka** [comment]
* About Q1

I think your data collection is good way. Always I do same way too. 

* About Q2

VAE can only reconstruct learned images. Therefore, even if you put a black image in the input, it will try to force the output of the track image. In this application, reconstruction image is not important. Only the latent representation output by the encoder is important.

* About Q3

Same answer as Q2. If only the same image is learned, only that image will be output. 

* About Q4 

I always use 50 to 100 epocks by 10000 images.

* About Q5 

Is it in VAE training? It is strange, always loss value should be decreasing. The following image is displayed on TensorBord. 
<img width="353" alt="スクリーンショット 2020-11-17 22 09 08" src="https://user-images.githubusercontent.com/1833346/99395263-86608a00-2923-11eb-890d-4a4e140a666a.png">
And my reconstruction image.
<img width="660" alt="スクリーンショット 2020-11-17 22 09 33" src="https://user-images.githubusercontent.com/1833346/99395301-8fe9f200-2923-11eb-88d6-4e3b8be24d31.png">

**Comment by Adnan-annan** [comment]
Thanks for your prompt reply. So i trained the VAE model. I had collected alot of test data also. and when i run the trained model on the test data results are very good. but when i run vae_viewer.ipynb it doesnt show similar results. probably results are not displayed properly in vae_viewer. I also validated the vae reconstruction image while training the jetracer for following the track. and the reconstruction was good. So i am certain now that vae is performing well and can concentrate on training the RL algorithm. Thanks for your help.
